controllers/clustergroupupgrade_controller.go

```diff
@@ -849,16 +849,24 @@ func (r *ClusterGroupUpgradeReconciler) updateConfigurationPolicyNameForCopiedPo
 
 func (r *ClusterGroupUpgradeReconciler) createNewPolicyFromStructure(
 	ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policy *unstructured.Unstructured) error {
+
+	if err := controllerutil.SetControllerReference(clusterGroupUpgrade, policy, r.Scheme); err != nil {
```

Comment:
Do we really need to SetControllerReference each time we are calling this function? To me, it seems that doing this upon creation is enough.

Reply:
This is actually the fix for the kuttl test failures, where updated objects get left behind causing subsequent testcases (with same CGU name) to fail. Before we use retrieved object to do update, that's why this is only needed in create case. Now that we do the real update with the desired policy object we just create, we have to set a few things like owner, resource version for the update to work.